x/config/spec.go

```diff
@@ -217,6 +223,27 @@ type TransportSpec struct {
 	// the first thing inside their BuildInbound.
 }
 
+// PeerListPreset defines a named preset for a peer list. Peer list presets
+// may be used by specifying a `with` key in the outbound configuration.
+//
+// 	http:
+// 	  with: mypreset
+type PeerListPreset struct {
+	Name string
+
+	// A function in the shape,
+	//
+	//  func(peer.Transport, *config.Kit) (peer.Chooser, error)
```

Comment:
It's `peer.Transport` similar to how the other build functions accept a `transport.Transport` and the transport author casts it into a specific type if necessary.

This can be a concrete function signature but see the comment below for why it's not: I want us to be able to support a more complex function signature in the future if needed, and we can't do that if this has a concrete type.